pkg/core/stateroot/module.go

```diff
@@ -114,6 +116,42 @@ func (s *Module) Init(height uint32, enableRefCount bool) error {
 	return nil
 }
 
+// CleanStorage removes all MPT-related data from the storage (MPT nodes, validated stateroots)
+// except local stateroot for the specified height and GC flag. This method is aimed to clean
+// outdated MPT data before state sync process can be started.
+func (s *Module) CleanStorage() error {
+	gcKey := []byte{byte(storage.DataMPT), prefixGC}
+	gcVal, err := s.Store.Get(gcKey)
+	if err != nil {
+		return fmt.Errorf("failed to get GC flag: %w", err)
+	}
+	b := s.Store.Batch()
```

Comment:
Are we gonna fit 10M of keys there?

Reply:
Yes, because we clean MPT data for genesis block only. I've added a check that this method is called at 0-height.